scrape/target.go

```diff
@@ -365,7 +365,7 @@ type bucketLimitAppender struct {
 
 func (app *bucketLimitAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels, t int64, h *histogram.Histogram, fh *histogram.FloatHistogram) (storage.SeriesRef, error) {
 	if h != nil {
-		if len(h.PositiveBuckets)+len(h.NegativeBuckets) > app.limit && h.Schema > histogram.ExponentialSchemaMax {
+		if len(h.PositiveBuckets)+len(h.NegativeBuckets) > app.limit && !histogram.IsExponentialSchema(h.Schema) {
```

Comment:
I think this is still not quite right. If we are currently at -4, we already want to return the error, but -4 is still a valid exponential schema. I'm afraid we have to also test for h.Schema being -4.

Reply:
That case is handled already on line 372.

Reply:
Note that the new if is an addition on the already existing code : https://github.com/prometheus/prometheus/blob/1c3f322f7832d6ebbba9dc3a13b398e03e014bcf/scrape/target.go#L367